src/Symfony/Component/HttpKernel/Event/ControllerEvent.php

```diff
@@ -22,6 +25,29 @@
  *
  * @author Bernhard Schussek <bschussek@gmail.com>
  */
-class ControllerEvent extends FilterControllerEvent
+class ControllerEvent extends KernelEvent
 {
+    private $controller;
+
+    public function __construct(HttpKernelInterface $kernel, callable $controller, Request $request, ?int $requestType)
+    {
+        parent::__construct($kernel, $request, $requestType);
+
+        $this->setController($controller);
+    }
+
+    /**
+     * Returns the current controller.
+     *
+     * @return callable
```

Comment:
let's use a real type instead? could require making either the class or the method final in 4.4